src/KafkaW/Consumer.cpp

```diff
@@ -13,264 +16,248 @@ static std::atomic<int> g_kafka_consumer_instance_count;
         rd_kafka_err2str((rd_kafka_resp_err_t)err));                           \
   }
 
-Consumer::Consumer(BrokerSettings BrokerSettings, ConsumerSettings CSettings)
-    : ConsumerBrokerSettings(BrokerSettings), Settings(CSettings) {
-  init();
+Consumer::Consumer(const BrokerSettings &BrokerSettings)
+    : ConsumerBrokerSettings(std::move(BrokerSettings)) {
+  ////C++
+  std::string ErrorString;
+  auto conf = RdKafka::Conf::create(RdKafka::Conf::CONF_GLOBAL);
+  conf->set("rebalance_cb", &RebalanceCallback, ErrorString);
+  conf->set("event_cb", &EventCallback, ErrorString);
+  conf->set("metadata.broker.list", ConsumerBrokerSettings.Address,
+            ErrorString);
+  ConsumerBrokerSettings.apply(conf);
+  this->KafkaConsumer = std::shared_ptr<RdKafka::KafkaConsumer>(
+      RdKafka::KafkaConsumer::create(conf, ErrorString));
+  if (!this->KafkaConsumer) {
+    LOG(Sev::Error, "can not create kafka consumer: {}", ErrorString);
+    throw std::runtime_error("can not create Kafka consumer");
+  }
+  ////C++__
```

Comment:
I am pretty sure that before we do the merge with master, we want the commented out code removed.